crypto-ffi/src/ciphersuite.rs

```diff
@@ -1,81 +1,167 @@
+//! Ciphersuites in bindings
+//!
+//! Both wasm-bindgen and uniffi support emitting enums, as long as they directly implement the enum;
+//! it doesn't work on newtypes around external enums. We therefore redefine the ciphersuites enum
+//! here with appropriate annotations such that it gets exported to all relevant bindings.
+
+use std::collections::HashSet;
+
 use core_crypto::prelude::{CiphersuiteName, MlsCiphersuite};
 #[cfg(target_family = "wasm")]
 use wasm_bindgen::prelude::*;
 
 use crate::{CoreCryptoError, CoreCryptoResult};
 
-#[derive(Debug, Clone, Copy, derive_more::From, derive_more::Into)]
+/// MLS ciphersuites.
+#[allow(non_camel_case_types)]
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, derive_more::TryFrom)]
+#[try_from(repr)]
+#[repr(u16)]
 #[cfg_attr(target_family = "wasm", wasm_bindgen, derive(serde::Serialize, serde::Deserialize))]
-pub struct Ciphersuite(CiphersuiteName);
+#[cfg_attr(not(target_family = "wasm"), derive(uniffi::Enum))]
+pub enum Ciphersuite {
+    /// DH KEM x25519 | AES-GCM 128 | SHA2-256 | Ed25519
+    #[default]
+    MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519 = 0x0001,
 
-#[cfg(not(target_family = "wasm"))]
-uniffi::custom_type!(Ciphersuite, u16, {
-    lower: |ciphersuite| (&ciphersuite.0).into(),
-    try_lift: |val| Ciphersuite::new(val).map_err(Into::into),
-});
+    /// DH KEM P256 | AES-GCM 128 | SHA2-256 | EcDSA P256
+    MLS_128_DHKEMP256_AES128GCM_SHA256_P256 = 0x0002,
 
-impl From<MlsCiphersuite> for Ciphersuite {
-    fn from(value: MlsCiphersuite) -> Self {
-        Self(value.into())
+    /// DH KEM x25519 | Chacha20Poly1305 | SHA2-256 | Ed25519
+    MLS_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519 = 0x0003,
+
+    /// DH KEM x448 | AES-GCM 256 | SHA2-512 | Ed448
+    MLS_256_DHKEMX448_AES256GCM_SHA512_Ed448 = 0x0004,
+
+    /// DH KEM P521 | AES-GCM 256 | SHA2-512 | EcDSA P521
+    MLS_256_DHKEMP521_AES256GCM_SHA512_P521 = 0x0005,
+
+    /// DH KEM x448 | Chacha20Poly1305 | SHA2-512 | Ed448
+    MLS_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448 = 0x0006,
+
+    /// DH KEM P384 | AES-GCM 256 | SHA2-384 | EcDSA P384
+    MLS_256_DHKEMP384_AES256GCM_SHA384_P384 = 0x0007,
+}
```

Comment:
Nit: is there a way to pick up these constant u16 values from somewhere else, some canonical place?
E.g.
```rust
MLS_256_DHKEMP384_AES256GCM_SHA384_P384 = MlsCiphersuite.MLS_256_DHKEMP384_AES256GCM_SHA384_P384.from()
```

I admit it would probably look more verbose, but at least we would not need to worry about keeping constants in sync.